sections/security/validation.md

```diff
@@ -0,0 +1,64 @@
+# Validate the incoming JSON schemas
+
+### One Paragraph Explainer
+Validation is about being very explicit on what payload our app is willing to accept and failing fast should the input deviates from the expectations. This minimizes attackers surface who can no longer tryout payloads with different structure, values and length. Practically it prevents attacks like DDOS (code is unlikely to fail when the input is well defined) and Insecure Deserialization (JSON contain no surprises ). Though validation can be coded or rely upon classes and types (TS, ES6 classes) the community seems to increasingly like JSON-based schemas as these allow declaring complex rules without coding and share the expectations with the frontend. JSON-schema is an emerging standard that is supported by many NPM libraries and tools (e.g. [jsonschema](https://www.npmjs.com/package/jsonschema), [POSTMAN](http://blog.getpostman.com/2017/07/28/api-testing-tips-from-a-postman-professional/)), [JOI](https://www.npmjs.com/package/joi) is also highly popular with sweet syntax. Typically JSON syntax can't cover all validation scenario and custom code or pre-baked validation frameworks like [validator.js](https://github.com/chriso/validator.js/) come in handy. Regardless of the chosen syntax, ensure to run the validation as early as possible - For example, by using Express middlewares that validates the request body before the request is passed to the route handler
```

Comment:
Like the fail fast comment 👍

Reply:
- replace DDOS with DOS
- fix extranous spacing in `(JSON contain no surprises )`